### PR TITLE
Pptx support

### PR DESCRIPTION
**Pull request by davidedigrande**
Addressing #253

I mostly followed the work done by @paskett for #281 

Tried with a bunch of random example pptx files found online to ensure wide compatibility.

Hope it's fine for you! This would be my first open source contribution, be kind :)

### CODE REVIEW

**Review by pmeier — COMMENTED** [review]
Thanks @davidedigrande for the PR. 

> This would be my first open source contribution, be kind :)

Welcome to the OSS community :rocket: I have a few high-level inline comments. Nothing major :slightly_smiling_face:

File: ragna/core/_document.py [hunk 1]
```diff
@@ -301,3 +301,31 @@ def extract_pages(self, document: Document) -> Iterator[Page]:
...
+    """Document handler for `.pptx` documents.
+
+    !!! info "Package requirements"
+
+        - [`python-pptx`](https://github.com/scanny/python-pptx)
+    """
+
+    @classmethod
+    def requirements(cls) -> list[Requirement]:
+        return [PackageRequirement("python-pptx>=0.6.23")]
```
[the remark this answers is not in the document]
Comment (davidedigrande) [reply ragna/core/_document.py:317]: Hi, there is not particular reason, it's just the version I did my tests with.
I wasn't able to find which version is the oldest compatible with the Ragna's suggested version of Python (3.9)
https://python-pptx.readthedocs.io/en/latest/user/install.html the only info I could find is `Python 2.6, 2.7, 3.3 or later`
I think we could probably be good without a pinned version

**Comment by davidedigrande** [comment]
Hi, I integrated @pmeier in the code and updated the test which was failing after applying the changes.
I hope we're all set now.

I don't have much time spare but I'll gladly keep contributing when I can, thanks!

File: ragna/core/_document.py [hunk 2]
```diff
@@ -301,3 +301,31 @@ def extract_pages(self, document: Document) -> Iterator[Page]:
...
+    def supported_suffixes(cls) -> list[str]:
+        return [".pptx"]
+
+    def extract_pages(self, document: Document) -> Iterator[Page]:
+        import pptx
+
+        document_pptx = pptx.Presentation(io.BytesIO(document.read()))
+        for number, slide in enumerate(document_pptx.slides, 1):
+            text = "\n\n".join(
+                shape.text for shape in slide.shapes if shape.has_text_frame and shape.text
```
Comment (pmeier) [inline ragna/core/_document.py:329]: Format

```suggestion
                shape.text 
                for shape in slide.shapes 
                if shape.has_text_frame and shape.text
```

**Review by pmeier — APPROVED** [review]
Thanks @davidedigrande we are almost set now. Just one lint issue I'll fix below. Plus, there is also an issue with the docker requirements, since we added something new. We have a bot that fixes this automatically after we merge this PR.

> I don't have much time spare but I'll gladly keep contributing when I can, thanks!

Awesome, happy to have you around :rocket:

File: ragna/core/_document.py [hunk 3]
```diff
@@ -301,3 +301,33 @@ def extract_pages(self, document: Document) -> Iterator[Page]:
...
+
+    def extract_pages(self, document: Document) -> Iterator[Page]:
+        import pptx
+
+        document_pptx = pptx.Presentation(io.BytesIO(document.read()))
+        for number, slide in enumerate(document_pptx.slides, 1):
+            text = "\n\n".join(
+                shape.text 
+                for shape in slide.shapes 
+                if shape.has_text_frame and shape.text
```
Comment (pmeier) [inline ragna/core/_document.py:331]: Trailing whitespace

```suggestion
                shape.text
                for shape in slide.shapes
                if shape.has_text_frame and shape.text
```

**Pull request merged**